time/src/serde/mod.rs

```diff
@@ -252,10 +252,18 @@ impl Serialize for Duration {
     fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
         #[cfg(feature = "serde-human-readable")]
         if serializer.is_human_readable() {
+            // Simply extracting sign and then calling `self.whole_seconds().abs()` will
+            // cause overflow on `i64::MIN` seconds, hence this kludge.
```

Comment:
Using `.unsigned_abs()` avoids the potential overflow while also being quite performant.

Reply:
Oh, I somehow missed it. Done.